services/telemetry/src/util.js

```diff
@@ -19,37 +19,37 @@ export function cleanupAfter(cleanup, limit, message) {
 
   let earlyCall = false;
   let earlyErr = null;
+  let cleanupObj = {};
 
   // Store if there was an error or not in the case that the finish
   // function is called before it is overrided below.
-  let finish = (err) => {
+  cleanupObj.finish = (err) => {
     earlyCall = true;
     earlyErr = err;
   };
 
-  // Resolves after the cleanup is finished.
-  let done = new Promise((resolve, reject) => {
-    // Call the finish function if the timeout is reached.
-    let waitTimeout = setTimeout(() => {
-      finish(Error(timeoutMessage));
-    }, limit);
-
-    // Cancel the timeout, cleanup, and mark the operation as
-    // complete.
-    finish = (err, result) => {
-      clearTimeout(waitTimeout);
-      cleanup();
-
-      if (err) reject(err);
-      else resolve(result);
-    };
-
-    if (earlyCall) finish(earlyErr);
-  });
-
-  // Async function to wait for the operation to be over and cleaned
-  // up.
-  let wait = async () => await done;
+  cleanupObj.wait = async () => {
+    // Resolves after the cleanup is finished.
+    let done = new Promise((resolve, reject) => {
+      // Call the finish function if the timeout is reached.
+      let waitTimeout = setTimeout(() => {
+        cleanupObj.finish(Error(timeoutMessage));
+      }, limit);
+
+      // Cancel the timeout, cleanup, and mark the operation as
+      // complete.
+      cleanupObj.finish = (err, result) => {
+        clearTimeout(waitTimeout);
+        cleanup();
+
+        if (err) reject(err);
+        else resolve(result);
+      };
+
+      if (earlyCall) cleanupObj.finish(earlyErr);
+    });
+    return await done;
```

Comment:
You should be able to do this:
```js
return await new Promise((resolve, reject) => { /* ... */ }
```

Reply:
I thought it was a bit easier to read factored out, but I can change it if you'd like.